sxg_rs/src/link.rs

```diff
@@ -31,21 +31,18 @@ pub(crate) async fn process_link_header(
         Lazy::new(|| vec!["", "anonymous"].into_iter().collect());
     match parse_link_header(value) {
         Ok(links) => {
-            let links: Vec<Link> = links
-                .into_iter()
-                .filter(|link| {
-                    link.params.iter().all(|(k, v)| {
-                        ALLOWED_PARAM.contains(k)
-                            && match *k {
-                                "rel" => matches!(v, Some(v) if ALLOWED_REL.contains(v.as_str())),
-                                "crossorigin" => {
-                                    matches!(v, Some(v) if ALLOWED_CROSSORIGIN.contains(v.as_str()))
-                                }
-                                _ => true,
+            let links = links.into_iter().filter(|link| {
+                link.params.iter().all(|(k, v)| {
+                    ALLOWED_PARAM.contains(k)
+                        && match *k {
+                            "rel" => matches!(v, Some(v) if ALLOWED_REL.contains(v.as_str())),
+                            "crossorigin" => {
+                                matches!(v, Some(v) if ALLOWED_CROSSORIGIN.contains(v.as_str()))
                             }
-                    })
+                            _ => true,
+                        }
                 })
-                .collect();
+            });
 
             let (mut preloads, allowed_alt_sxgs) = links
                 .into_iter()
```

Comment:
You can probably get rid of the `into_iter()` here, now that you removed the `collect()` above.

Reply:
Done